packages/core/src/new-api/internal/deployer.ts

```diff
@@ -52,12 +57,20 @@ export class Deployer {
     );
   }
 
-  public async deploy(
-    ignitionModule: IgnitionModule,
+  public async deploy<
+    ModuleIdT extends string,
+    ContractNameT extends string,
+    IgnitionModuleResultsT extends IgnitionModuleResult<ContractNameT>
+  >(
+    ignitionModule: IgnitionModule<
+      ModuleIdT,
+      ContractNameT,
+      IgnitionModuleResultsT
+    >,
     deploymentParameters: DeploymentParameters,
     accounts: string[],
     defaultSender: string
-  ): Promise<DeploymentResult> {
+  ): Promise<DeploymentResult<ContractNameT, IgnitionModuleResultsT>> {
     await validateStageTwo(
```

Comment:
This is something you can take over if you want @kanej.

Here the validate functions are throwing an error that doesn't give you access to the futureId that caused the error.

Ideally, the validation function would return all the errors. A quick fix is changing the error they throw so that they include the futureId. We can do the proper fix later.

Reply:
I am tackling this now.

I am going to alter the validate stage two to return [futureId, validationErrorMessage] pairs (or some such) that can then be returned as a result if there are any validation failures.

Reply:
Alright, I backed out of the larger change and did as you suggested.

Stage two now returns either a `ValidationErrorDeploymentResult` or null, and deploy returns the error result if detected.

Reply:
I have added an issue to expand out the number of validation messages returned: https://github.com/NomicFoundation/ignition/issues/424